calcom/cal.com
# feat: daily webhooks

Fixes: https://github.com/calcom/cal.com/issues/12185

How to test ? 

Make sure you have DAILY_API_KEY, DAILY_SCALE_PLAN="true"

1) ngrok http 3000 and get the ngrok url
2) Create a webhook by 

```
curl --location --request POST 'https://daily.co/api/v1/webhooks' \
--header 'Authorization: Bearer $token
' \
--header 'Content-Type: text/plain' \
--data-raw '{
    "url": "<ngrok_url>/api/recorded-daily-video”,
    "eventTypes": [ "recording.ready-to-download"]
}'

```
**Note**:- Make sure to delete the webhook after testing

3) You would get hmacSecret in the response and put this value in DAILY_WEBHOOK_SECRET env variable.

4) Create a booking with daily video as location

5) Join the daily video meeting link and click on start recording and then stop recording after few seconds.

6) Organizer and Attendees would receive an email with the recording link

<img width="1014" alt="Screenshot 2023-11-18 at 2 14 04 PM" src="https://github.com/calcom/cal.com/assets/53316345/fcdc53a0-256d-4d75-9847-f48cd61177e9">


Instructions after deploying 

 Create webhook with url https://app.cal.com/api/recorded-daily-video

```
curl --location --request POST 'https://daily.co/api/v1/webhooks' \
--header 'Authorization: Bearer $token
' \
--header 'Content-Type: text/plain' \
--data-raw '{
    "url": "https://app.cal.com/api/recorded-daily-video”,
    "eventTypes": [ "recording.ready-to-download"]
}'


```
